cmake/compile_definitions/common.cmake

```diff
@@ -3,7 +3,25 @@
 
 list(APPEND SUNSHINE_COMPILE_OPTIONS -Wall -Wno-sign-compare)
 # Wall - enable all warnings
+# Werror - treat warnings as errors
+# Wno-maybe-uninitialized/Wno-uninitialized - disable warnings for maybe uninitialized variables
 # Wno-sign-compare - disable warnings for signed/unsigned comparisons
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    # GCC specific compile options
+
+    # GCC 12 and higher will complain about maybe-uninitialized
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 12)
+        list(APPEND SUNSHINE_COMPILE_OPTIONS -Wno-maybe-uninitialized)
+    endif()
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    # Clang specific compile options
+
+    # Clang doesn't actually complain about this this, so disabling for now
+    # list(APPEND SUNSHINE_COMPILE_OPTIONS -Wno-uninitialized)
+endif()
```

Comment:
Nothing enabled currently, but keeping the else block as a placeholder for future Clang specific items.